### PR TITLE
Remove util_make_unique

### PR DESCRIPTION
**Pull request by tautschnig**
With the move to C++ 17 we can use std::make_unique instead.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by esteffin — APPROVED** [review]

**Review by NlightNFotis — APPROVED** [review]

**Review by kroening — APPROVED** [review]

**Review by thomasspriggs — APPROVED** [review]

**Review by TGWDB — APPROVED** [review]

**Review by peterschrammel — APPROVED** [review]

**Comment by thomasspriggs** [comment]
@martin-cs This seems to need your review due to a change in `/src/json-symtab-language/`

**Comment by thomasspriggs** [comment]
> @martin-cs This seems to need your review due to a change in `/src/json-symtab-language/`

That should be resolved by the addition of extra code owners in https://github.com/diffblue/cbmc/pull/8087


I think we still need review from one of @qinheping @feliperodri or @remi-delmas-3000 due to the update in `src/goto-synthesizer/`

**Review by feliperodri — APPROVED** [review]
Changes in `goto/synthesizer` LGTM.

**Pull request merged**
